// animescraper/index.js
const discord = require("discord.js");
const fetch = require("node-fetch");
const cheerio = require("cheerio");


const searchUrl = 'https://myanimelist.net/anime.php?q=';
const animeUrl = 'https://myanimelist.net/anime/';

async function searchAnime(query) {
    return fetch(`${searchUrl}${query}`)
    .then(res => res.text())
    .then(body => {
        const animes = [];
        const $ = cheerio.load(body);
        $('a.hoverinfo_trigger' && 'a.fw-b').each(function(i, element) {
            const $element = $(element);
            const atitle = $element.text();
            const alink = $element.attr('href').slice(30);
            const anime = {
                title: `${atitle}`,
                link: `${alink}`
            }
            animes.push(anime);
            return false;
        });
    
        return animes;
    
        
    });
}


async function getAnime(alink) {
    return fetch(`${animeUrl}${alink}`)
        .then(res => res.text())
        .then(body => {
            const $ = cheerio.load(body);
            
            const title = $('.h1 span').text().trim();
            const ranked = $('.ranked strong').text().trim();
            const popularity = $('.popularity strong').text().trim();
            const score = $('.score').text().trim();
            const image = $('.ac').attr('src');
            const desc = $('span[itemProp="description"]').text().trim();
            const description = desc;
            let episode = String;

            $('.spaceit span.dark_text').each(function(i, element) {
                const $element = $(element);
                const $ep = $element.parent().text().trim();
                episode = $ep;
                return false;
            });

            episodes = episode.slice(12, 17);

            infoarray = {
                title,
                ranked,
                popularity,
                score,
                image,
                description,
                episodes
            }

            return infoarray;

        });

}




module.exports = {
    searchAnime,
    getAnime,
    animeUrl
};

module.exports.help = {
    name: "animescraper.js"
}
